### PR TITLE
Issue 3658: FlowId and Request Id in InvalidEventNumber Wirecommand

### LINKED ISSUE #3658: Confusion between flowId and EventNumber in InvalidEventNumber
**Problem description**
The wire command InvalidEventNumber has a member which is called eventNumber but it is actually the flowId. It is consistently treated correctly, but the parameter name is very confusing.

**Problem location**
WireCommands.InvalidEventNumber

**Suggestions for an improvement**
Rename the field to flowId

### PR DESCRIPTION
**Pull request by kotlasaicharanreddy**
Signed-off-by: SaiCharan <kotlasaicharan@yahoo.com>

**Change log description**  
`eventNumber` in InvalidEventNumber is actually returning a flowId instead of an eventNumber. It is correct functionality wise but the name suggests otherwise. 
Changed the `eventNumber` variable name to `requestId` to avoid confusion between flowId and eventnumber and added a new field `eventNumber`.

**Purpose of the change**  
Fixed #3658

**What the code does**  
- Renamed `eventNumber` to requestId in InvalidEventNumber Wire Command
- Added a new field `eventNumber`.
- Added a unitTest `**InvalidEventNumberV15**` to make sure that it is backward compatible.

**How to verify it**  
- All tests must pass.
- `InvalidEventNumberV15` should always pass

### CODE REVIEW

File: client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ RuntimeException handleUnexpectedReply(Reply reply, String expectation) {
         } else if (reply instanceof InvalidEventNumber) {
             InvalidEventNumber ien = (InvalidEventNumber) reply;
             throw Exceptions.sneakyThrow(new ConnectionFailedException(ien.getWriterId() + 
-                    " Got stale data from setupAppend on segment " + segmentId + " for ConditionalOutputStream. Event number was " + ien.getEventNumber()));
+                    " Got stale data from setupAppend on segment " + segmentId + " for ConditionalOutputStream. Flow number was " + ien.getRequestId()));
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java:152]: Can we log both?

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java [hunk 2]
```diff
@@ -362,6 +362,7 @@ public static final class InvalidEventNumber implements Reply, WireCommand {
         final WireCommandType type = WireCommandType.INVALID_EVENT_NUMBER;
         final UUID writerId;
         final long eventNumber;
+        final long requestId;
```
Comment (tkaitchuck) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java:365]: This is not backward compatible. New fields need to go at the end. So the event number field should be renamed `requestId` and a new `eventNumber` field should be added to the end.

**Review by tkaitchuck — CHANGES_REQUESTED** [review]

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java [hunk 3]
```diff
@@ -373,15 +375,17 @@ public void process(ReplyProcessor cp) {
         public void writeFields(DataOutput out) throws IOException {
             out.writeLong(writerId.getMostSignificantBits());
             out.writeLong(writerId.getLeastSignificantBits());
+            out.writeLong(requestId);
             out.writeLong(eventNumber);
```
Comment (tkaitchuck) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java:379]: This needs to go last

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java [hunk 4]
```diff
@@ -373,15 +375,17 @@ public void process(ReplyProcessor cp) {
...
             out.writeUTF(serverStackTrace);
         }
 
         public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
             UUID writerId = new UUID(in.readLong(), in.readLong());
+            long requestId = in.readLong();
             long eventNumber = in.readLong();
```
Comment (tkaitchuck) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java:386]: This needs to go last with an "available" check just like serverStackTrace.

File: client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java (hunk 1 above)
Comment (kotlasaicharanreddy) [reply client/src/main/java/io/pravega/client/segment/impl/ConditionalOutputStreamImpl.java]: Done. Logging both eventNumber and requestId

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java (hunk 3 above)
Comment (kotlasaicharanreddy) [reply shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java]: Done. Moved it to last to make backward compatible.

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java (hunk 4 above)
Comment (kotlasaicharanreddy) [reply shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java]: Done. Moved it to last to make backward compatible.

**Review by tkaitchuck — APPROVED** [review]

**Review by SrishT — APPROVED** [review]
LGTM

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**